localstack-core/localstack/testing/pytest/stepfunctions/utils.py

```diff
@@ -587,10 +611,18 @@ def create_and_record_mocked_execution(
         sfn_snapshot,
         definition,
         state_machine_name=state_machine_name,
+        state_machine_type=state_machine_type,
     )
-    execution_arn = launch_and_record_mocked_execution(
-        target_aws_client, sfn_snapshot, state_machine_arn, execution_input, test_name
-    )
+    if state_machine_type == StateMachineType.EXPRESS:
```

Comment:
There are STANDARD and EXPRESS state machines. LocalStack wants to enable the mock feature for both. However, when using the `StartSyncExecution` API action, only STANDARD state machines can be executed, otherwise an error is raised. This utility seems to be limited to recording the executions through StartExecution only, as it invokes the asynchronous await logic. 
What do you think about reverting the changes in this function, and adding the snapshotting features for mocked executions through StartSyncExecution in your launch_and_record_mocked_sync_execution? Also considering create_and_record_express_sync_execution. This could also allow us to test for failures if a STANDARD with mocking through StartSyncExecution.

Reply:
Thank you, I left a new comment with a minor. 
Regarding the use of STANDARD and EXPRESS state machine, yes I would agree [I see `StartSyncExecution` should only be available for `EXPRESS` workflows](https://docs.aws.amazon.com/step-functions/latest/apireference/API_StartSyncExecution.html).